Expensify/App
# Ensure Staging environment badge is shown for beta builds on iOS and Android

@roryabraham will you please review this?

### Details
Since we use the same builds for staging and production, both staging and production builds on iOS and Android use the production `.env`, so we need another way to differentiate whether the app is staging or production. iOS allows us to check if the app is a TestFlight build, so this PR adds a nativeModule to check that on startup. For the Android check, there isn't an easy way to do this, so we have to query the play store and grab the current version from the html.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/2597

### Tests
1. Run the app on dev on all platforms and sign out, confirm that you see a `dev` badge on the sign-in screen logo. On desktop, confirm you see a `dev` badge for the doc icon.
2. Login to any account, confirm after logging in you see a `dev` badge on the main app screen next to `Chats`
3. Confirm that after logging out, the proper icon shows on staging and dev
3. On desktop, run `npm run desktop-build-staging`, then open the Expensify.cash app inside `dist/mac`. Confirm the icon has the `stg` badge.
4. Confirm on the sign-in screen the icon has a `stg` badge. Sign in and confirm the `stg` badge shows up next to `Chats`
5. Update your `.env` file to include `ENVIRONMENT=PROD` and run the app on web/ios, confirm you see no badges on the sign-in screen and in the app
6. Update the return statement in `EnvironmentChecker.m` to `return [NSNumber numberWithBool:isRunningTestFlightBeta];` and set `ENVIRONMENT=PROD` in your `.env` file. Confirm that when running the app, you see the `STG` badge (we'll test this live once its merged to confirm it shows up properly on the actual TestFlight)
7. On Android, set the ENVIRONMENT to PROD and run the app. Confirm that you see the STG badge
8. Update the `CONST.PLAY_STORE_URL` constant inside setEnvironment/index.android.js to be a nonsense URL. Run the app again, confirm no badge is shown
9. Reset the `CONST.PLAY_STORE_URL` value so it is correct again, then update the `version` value inside `package.json` and `package-lock.json` to the `Current Version` from https://play.google.com/store/apps/details?id=com.expensify.chat&hl=en
10. Run the Android app, confirm no badge shows

### QA Steps
1. Confirm that when running the app on staging, an `STG` badge appears on the sign-in screen icon, and in the main app next to `Chats` (after login) on Desktop, iOS, and Web
    2. Confirm that on desktop, the desktop icon shows with the `STG` badge
2. Sign out of the app, confirm STG still shows on the login screen icon
2. Confirm that when running the app on prod (all platforms), there are no badges

### Tested On
